src/internal.rs

```diff
@@ -203,12 +204,37 @@ pub struct Rav1dContext_intra_edge {
     pub tip_sb64: [EdgeTip; 64],
 }
 
+pub(crate) struct Rav1dContextTaskThread {
+    /// Thread join handle, if this task is on a worker thread. The main thread
+    /// task does not contain a handle.
+    pub(crate) handle: Option<pthread_t>,
+    /// Data shared between the main thread and a worker thread.
+    pub(crate) thread_data: Arc<Rav1dTaskContext_task_thread>,
+}
+
+impl Rav1dContextTaskThread {
+    pub(crate) fn new(
+        handle: Option<pthread_t>,
+        thread_data: Arc<Rav1dTaskContext_task_thread>,
+    ) -> Self {
+        Self {
+            handle,
+            thread_data,
+        }
+    }
```

Comment:
If the constructor isn't doing anything and the fields aren't private, I think just directly constructing the type is better than calling an `fn new`, since you get things like field names, too.

Reply:
I prefer a method here, personally, but it's not a big deal. I kind of wanted to make these fields private but we don't really need to here.